lib/routes/redirect.js

```diff
@@ -13,9 +14,11 @@ function actionToPathname(action) {
     return 'force_auth';
   } else if (action === 'signin') {
     return 'signin';
+  } else if (action === undefined) {
+    return '';
   }
 
-  return '';
+  throw AppError.incorrectRedirect();
```

Comment:
This error actually is for when the `redirect_uri` doesn't match with what was registered for the client.